web/src/components/filtersform.js

```diff
@@ -276,15 +280,38 @@ class FiltersForm extends React.Component {
       {params.get('approvals') && this.props.showChangeParams ? ', approvals ' + params.get('approvals') : ''}
       {params.get('exclude_approvals') && this.props.showChangeParams ? ', excluded approvals named ' + params.get('exclude_approvals') : ''}
       {params.get('state') && this.props.showChangeParams ? ', change state ' + params.get('state') : ''}
+      {params.get("project_definition") ? ", change project_definition " + params.get("project_definition") : ""}
       {'.'}
     </b>
     return resume
   }
+  handleProjectDefState = (e) => {
+    const selected = e.target.attributes.value.value;
+    this.setState({ project_definition: selected });
+  };
+
+  renderProjectDefs = () => {
```

Comment:
While this may work, I don't think storing the dom elements in the state is a good design. Perhaps it would be better to simply save the response with setState here, and do the rendering in the render function below.